3-UsingAPIs/Week1/homework/ex1-johnWho.js

```diff
@@ -8,22 +8,23 @@ Rewrite this function, but replace the callback syntax with the Promise syntax:
 - If the Promise `rejects`, pass an error as the argument to reject with: "You 
   didn't pass in a first name!"
 ------------------------------------------------------------------------------*/
-// TODO see above
-const getAnonName = (firstName, callback) => {
-  setTimeout(() => {
-    if (!firstName) {
-      callback(new Error("You didn't pass in a first name!"));
-      return;
-    }
 
-    const fullName = `${firstName} Doe`;
-
-    callback(fullName);
-  }, 1000);
+const getAnonName = (firstName) => {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      if (!firstName) {
+        reject(new Error("You didn't pass in a first name!"));
+        return;
+      }
+      resolve(`${firstName} Doe`);
+    }, 1000);
+  });
 };
 
 function main() {
-  getAnonName('John', console.log);
+  getAnonName('John')
+    .then((result) => console.log(result))
+    .catch((error) => console.log(error));
```

Comment:
Rather than printing the whole error object, better to just print the error message.

```suggestion
    .catch((error) => console.log(error.message));
```